lib/futurism/channel.rb

```diff
@@ -15,17 +16,17 @@ def subscribed
     end
 
     def receive(data)
-      resources = data.fetch_values("signed_params", "sgids") { |key| [nil] }.transpose
-
+      resources = data.fetch_values("signed_params", "sgids", "signed_controllers") { |_key| Array.new(data["signed_params"].length, nil) }.transpose
       new_env = connection.env.merge(ApplicationController.renderer.instance_variable_get(:@env))
```

Comment:
do we need to use the resolved `controller` here as well? This is basically a monkey patch to please Devise etc.

Reply:
I wasn't sure exactly what this was doing, so maybe?

Since we don't have full integration tests, it appears, it's hard to know if this whole PR has the intended consequences.

If we do use `controller` then we should likely a) move all of this instantation within the loop, and b) memoize so that we don't instantiate the same controllers many many times.